src/store/modules/interactiveMap.ts

```diff
@@ -0,0 +1,253 @@
+import Vue from "vue";
+import axios from "axios";
+import { AxiosResponse } from "axios";
+import * as settings from "@/utils/settings";
+import { OrganismItem } from "./organisms";
+import { ModelItem } from "./models";
+
+export interface Metabolite {
+  id: string;
+  naem: string;
+  compartmentId: string;
+  stoichiometry: number;
+}
+
+export interface Reaction {
+  id: string;
+  name: string;
+  reactionString: string;
+  metabolites: Metabolite[];
+  lowerBound?: number; // Used when editing bounds
+  upperBound?: number; // Used when editing bounds
+}
+
+export interface Card {
+  uuid: string;
+  name: string;
+  organism: OrganismItem;
+  model: ModelItem;
+  isLoadingModel: boolean;
+  fullModel: ModelItem;
+  method: string;
+  // Design card fields
+  objective: {
+    reaction: Reaction;
+    maximize: boolean;
+  };
+  reactionAdditions: Reaction[];
+  reactionKnockouts: Reaction[];
+  geneKnockouts: any[];
+  editedBounds: Reaction[];
+  // Data-driven card fields
+  experiment: any;
+  condition: any;
+  conditionData: any;
+  conditionWarnings: any[];
+  conditionErrors: any[];
+  // General simulation fields
+  isSimulating: boolean;
+  hasSimulationError: boolean;
+  hasLoadModelError: boolean;
+  growthRate: number;
+  fluxes: number;
+}
+
+export default {
+  namespaced: true,
+  state: {
+    cards: [] as Card[]
+  },
+  mutations: {
+    addCard(state, card) {
+      state.cards.push(card);
+    },
+    removeCard(state, card) {
+      state.cards.splice(state.cards.indexOf(card), 1);
+    },
+    updateCard(state, { uuid, props }) {
+      const card = state.cards.find(c => c.uuid === uuid);
+      Object.keys(props).forEach(key => {
+        card[key] = props[key];
```

Comment:
I don't think Vuex can detect these (and other) deep object changes (e.g. to re-run dependent `computed` properties on components)

Recommendation here: https://vuex.vuejs.org/guide/mutations.html#mutations-follow-vue-s-reactivity-rules

Reply:
That's indeed the case when you add new properties to an object. Here we're just modifying existing properties though, and that should still be detected by deep watchers. Regardless, it's probably good practice to just use `Vue.set` in general, so I'll update this - thanks.